media-source/MediaSource-addSourceBuffer_infinite-calling.html

```diff
@@ -0,0 +1,55 @@
+<!doctype html>
+<html>
+<head>
+  <meta charset='utf-8'>
+  <title>Infinite calling MediaSource#addSourceBuffer</title>
+  <script src="/resources/testharness.js"></script>
+  <script src="/resources/testharnessreport.js"></script>
+</head>
+<body>
+<div id="log"></div>
+
+<script>
+var video = document.createElement('video');
+var mimes = ['video/webm; codecs="vorbis,vp8"', 'video/mp4'];
+
+//check the browser supports the MIME used in this test
+function isTypeSupported(mime) {
+    if(!MediaSource.isTypeSupported(mime)) {
+        this.step(function() {
+            assert_unreached("Browser doesn't support the MIME used in this test: " + mime);
+        });
+        this.done();
+        return false;
+    }
+    return true;
+}
+function testMedia(mime) {
+    async_test(function(t) {
+        if(!isTypeSupported.bind(t)(mime)) {
+            return;
+        }
+        var mediaSource = new MediaSource();
+        mediaSource.addEventListener('sourceopen', function(e) {
+            var sourceBuffer = null;
+            t.step(function() {
+                assert_throws({name : 'QuotaExceededError'},
+                              function() {
+                                  while(true) {
+                                      sourceBuffer = mediaSource.addSourceBuffer(mime);
+                                  }I
+                              },
+                              'Create infinite sourceBuffers');
+            });
+            t.done();
+        }, false);
+        video.src = window.URL.createObjectURL(mediaSource);
+    }, 'MediaSource#addSourceBuffer (' + mime + ') : ' +
+       'If the user agent can\'t handle any more SourceBuffer objects then throw a QUOTA_EXCEEDED_ERR exception and abort these steps');
+}
+mimes.forEach(function(mime) {
```

Comment:
mediasource-util.js provides:
MediaSourceUTIL.{AUDIO_ONLY,VIDEO_ONLY,AUDIO_VIDEO}_TYPE mime types that are precalculated as ones that have passed isTypeSupported. Using that, this test file can be significantly simplified.